core/src/SYCL/Kokkos_SYCL_Parallel_Reduce.hpp

```diff
@@ -201,7 +201,7 @@ class ParallelReduce<FunctorType, Kokkos::RangePolicy<Traits...>, ReducerType,
         Kokkos::Impl::FunctorValueJoin<ReducerTypeFwd, WorkTagFwd>;
     using ValueOps = Kokkos::Impl::FunctorValueOps<FunctorType, WorkTag>;
 
-    auto selected_reducer = ReducerConditional::select(functor, reducer);
+    auto& selected_reducer = ReducerConditional::select(functor, reducer);
```

Comment:
Yeah, let's see if
```suggestion
    const auto& selected_reducer = ReducerConditional::select(functor, reducer);
```
works.